service/src/main/java/greencity/service/EcoNewsServiceImpl.java

```diff
@@ -317,7 +315,7 @@ public EcoNewsDto findDtoByIdAndLanguage(Long id, String language) {
     public void delete(Long id, UserVO user) {
         EcoNewsVO ecoNewsVO = findById(id);
         if (user.getRole() != Role.ROLE_ADMIN && !user.getId().equals(ecoNewsVO.getAuthor().getId())) {
-            throw new BadRequestException(ErrorMessage.USER_HAS_NO_PERMISSION);
+            throw new UserHasNoPermissionToAccessException(ErrorMessage.USER_HAS_NO_PERMISSION);
```

Comment:
This code was here, I just changed the exception that is thrown. Transferring it to the SecurityConfig will not be correct because checking whether you are the owner of this news is not a SecurityConfig task

Reply:
According to the mentor's answer, your variant is correct.